emc/estimator.py

```diff
@@ -0,0 +1,52 @@
+"""A SHORELine-like algorithm for the realignment of dMRI data."""
+import attr
+import numpy as np
+from pathlib import Path
+
+
+@attr.s(slots=True)
+class EddyMotionEstimation:
+    """Estimate rigid-body head-motion and distortions derived from eddy-currents."""
+
+    datapath = attr.ib(default=None)
+    """Path to the input DWI dataset."""
+    gradients = attr.ib(default=None)
+    """A 2D numpy array of the gradient table in RAS+B format."""
```

Comment:
Why is the data a path and the gradients an array? Maybe we could allow either of these as input for both of these?

Reply:
I'm still thinking about this one, and go either one way or another (but not the way it is now, like you said).

If we want to enjoy the benefits of HDF5, just a path would be the best option. Otherwise, yes, we want to have all-arrays probably.

Reply:
actually, with the new dwi object, the idea was to remove all of these.